### PR TITLE
Add `timestep_spacing` and `steps_offset` to schedulers

### PR DESCRIPTION
**Pull request by pcuenca**
Refer to previous discussions in #3929.

### CODE REVIEW

File: src/diffusers/configuration_utils.py [hunk 1]
```diff
@@ -551,8 +551,9 @@ def to_json_saveable(value):
             return value
 
         config_dict = {k: to_json_saveable(v) for k, v in config_dict.items()}
-        # Don't save "_ignore_files"
+        # Don't save "_ignore_files" or "_use_default_values"
         config_dict.pop("_ignore_files", None)
+        config_dict.pop("_use_default_values", None)
```
Comment (pcuenca) [inline src/diffusers/configuration_utils.py:557]: Thanks! I thought about testing this but never did it.

File: src/diffusers/configuration_utils.py [hunk 2]
```diff
@@ -508,7 +508,8 @@ def extract_init_dict(cls, config_dict, **kwargs):
         hidden_config_dict = {k: v for k, v in original_dict.items() if k not in init_dict}
 
         # 8. Take note of the parameters that were not present in the loaded config
-        hidden_config_dict["_use_default_values"] = expected_keys - set(init_dict)
+        if len(expected_keys - set(init_dict)) > 0:
+            hidden_config_dict["_use_default_values"] = expected_keys - set(init_dict)
```
Comment (pcuenca) [inline src/diffusers/configuration_utils.py:512]: 👍

**Review by patrickvonplaten — APPROVED** [review]

File: tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py [hunk 3]
```diff
@@ -304,7 +304,8 @@ def test_stable_diffusion_pix2pix_zero_ddpm(self):
         image_slice = image[0, -3:, -3:, -1]
 
         assert image.shape == (1, 64, 64, 3)
-        expected_slice = np.array([0.4861, 0.5053, 0.5038, 0.3994, 0.3562, 0.4768, 0.5172, 0.5280, 0.4938])
+        print(torch.from_numpy(image_slice.flatten()))
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py:307]:
```suggestion
```

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 4]
```diff
@@ -134,6 +141,8 @@ def __init__(
         dynamic_thresholding_ratio: float = 0.995,
         clip_sample_range: float = 1.0,
         sample_max_value: float = 1.0,
+        timestep_spacing: str = "leading",
+        steps_offset: int = 1,
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddpm.py:145]:
```suggestion
        steps_offset: int = 0,
```

Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_ddpm.py:145]: Actually shouldn't that default to 0?

File: tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py [hunk 5]
```diff
@@ -304,7 +304,7 @@ def test_stable_diffusion_pix2pix_zero_ddpm(self):
         image_slice = image[0, -3:, -3:, -1]
 
         assert image.shape == (1, 64, 64, 3)
-        expected_slice = np.array([0.4861, 0.5053, 0.5038, 0.3994, 0.3562, 0.4768, 0.5172, 0.5280, 0.4938])
+        expected_slice = np.array([0.4833, 0.5052, 0.5034, 0.4022, 0.3577, 0.4766, 0.5176, 0.5288, 0.4942])
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py:307]:
```suggestion
        expected_slice = np.array([0.4861, 0.5053, 0.5038, 0.3994, 0.3562, 0.4768, 0.5172, 0.5280, 0.4938])
```

**Pull request merged**